discovery/service_test.go

```diff
@@ -383,6 +411,11 @@ func createTestCtx(startHeight uint32) (*testCtx, func(), error) {
 	notifier := newMockNotifier()
 	router := newMockRouter(startHeight)
 
+	db, cleanUpDb, err := makeTestDB()
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
This is context constructor, `db` should be present after exiting from function.